app/src/androidTest/java/com/example/bespinaf/a2d2/test_denies_permissions.java

```diff
@@ -52,27 +44,20 @@ public void setUp(){
         device = UiDevice.getInstance(getInstrumentation());
     }
 
-    @Test
+    @Test // Checking that the activity has loaded
     public void hasActivity(){
         assertNotNull(activity);
     }
 
-    @Test
-    public void hasRulesText(){
-        Context appContext = activity.getApplicationContext();
-        String rulesText = appContext.getResources().getString(R.string.a2d2_rules_text);
-        final UiObject rulesTextView = device.findObject(new UiSelector().text(rulesText));
-        assertTrue(rulesTextView.exists());
-    }
-
-    @Test
+    @Test // Makes sure that the Agree button is on the screen
     public void hasButton(){
         Button testButtonAgree = activity.findViewById(R.id.button_rules_agree);
         assertNotNull(testButtonAgree);
     }
 
-    @Test
+    @Test //Tests the user can see an error pop-up when they deny permissions
```

Comment:
Update comment to reflect standard practice. Example:

/**
 * Lorem ipsum
 *
 * @param derp
 */
public void String canUserAgree(String derp) {
}